Backpack-SwiftUI/Skeleton/Classes/BPKSkeleton.swift

```diff
@@ -0,0 +1,103 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+import Backpack_Common
+
+public struct BPKSkeleton: View {
+    private var type: BPKSkeleton.ViewType = .image
+    private var size: BPKSkeleton.Size = .default
+    private var style: BPKSkeleton.Style = .default
+    private var color: Color = Color(.skyGrayTint06.darkVariant(.blackTint02))
+
+    @State private var phase: CGFloat = 0
+    let duration = 1.0
+    let delay = 0.2
+    
+    public init(
+        type: BPKSkeleton.ViewType,
+        size: BPKSkeleton.Size = .default,
+        style: BPKSkeleton.Style = .default
+    ) {
+        self.type = type
+        self.style = style
+        self.size = size
+
+    }
+    
+    public var body: some View {
+        switch type {
+        case .image:
+            RoundedRectangle(cornerRadius: style.cornerRadius)
+               .fill(color)
+               .frame(width: size.image.width, height: size.image.height)
+               .clipped()
+            
+        case .headline:
+            RoundedRectangle(cornerRadius: BPKCornerRadius.xs)
+               .fill(color)
+               .frame(width: size.headline.width, height:  size.headline.height)
+               .clipped()
+            
+        case .circle:
+            Circle()
+               .fill(color)
+               .frame(width: size.circle.width, height:  size.circle.height)
+               .clipped()
+            
+        case .bodytext:
+            VStack(alignment: .leading, spacing: BPKSpacing.sm.value * 2.5) {
```

Comment:
Would a normal BPKSpacing.md be sufficient? (it's x2 of BPKSpacing.sm)

Reply:
Nope, the spacing of UI design is 10, not 8, I can change it to `10`
![image](https://user-images.githubusercontent.com/13347837/186354388-d324b825-73a1-4212-84c2-f35f21d6ceb9.png)